libraries/AP_Compass/AP_Compass_AK8963.cpp

```diff
@@ -189,18 +167,19 @@ bool AP_Compass_AK8963::init()
 
     /* register the compass instance in the frontend */
     _compass_instance = register_compass();
-    set_dev_id(_compass_instance, AP_COMPASS_TYPE_AK8963_MPU9250);
+
+    set_dev_id(_compass_instance, _bus->get_dev_id());
```

Comment:
> the bus would need to access a method from the backend. Which is, I think even worse. Do you see a way that in a clean fashion?

I'm not sure why it's worse having the AP_AK8963_SerialBus_MPU9250 to call the set_dev_id() method when it's correctly configured. But it's maybe just the "_bus" naming that is weird.  I'm ok with letting this as is.